src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -1931,6 +1939,74 @@ public List<String> adminApprovalFields() {
 		}
 		return new ArrayList<>();
 	}
+
+
+	/**
+	 * Updates the user profile with version 2.
+	 *
+	 * @param request   The request containing profile update data.
+	 * @param userToken The user token for authentication.
+	 * @param authToken The authentication token.
+	 * @param rootOrgId The root organization ID.
+	 * @return SBApiResponse object containing the response of the profile update.
+	 * @throws Exception if any error occurs during the profile update process.
+	 */
+	@Override
+	public SBApiResponse profileUpdateV2(Map<String, Object> request, String userToken, String authToken, String rootOrgId) throws Exception {
+		// Extracting the request map from the input request
+		Map<String, Object> requestMap = (Map<String, Object>) request.get(Constants.REQUEST);
+		// Validating the token from the request map
+		if (validateToken((String) requestMap.get("token"))) {
```

Comment:
This method throws exception back to controller - which is wrong
Also it should not return null.

Refer the other method implementations - service methods should return only SBApiResponse -- with OK or with error messages.

Reply:
1.Handled with try and catch and return the SBIResponse with error message.
2. Removed the exception sent to the controller class.